src/ImageManipulation.php

```diff
@@ -597,52 +597,13 @@ public function getImageBackend()
     }
 
     /**
-     * Get the dimensions of this Image.
-     *
-     * @param string $dim One of the following:
-     *  - "string": return the dimensions in string form
-     *  - "array": it'll return the raw result
-     *  - 0: return the height
-     *  - 1: return the width
-     * @return string|int|array|null
-     */
-    public function getDimensions($dim = "string")
-    {
-        if (!$this->getIsImage()) {
-            return null;
-        }
-
-        $content = $this->getString();
-        if (!$content) {
-            return null;
-        }
-
-        // Get raw content
-        $size = getimagesizefromstring($content);
-        if ($size === false) {
-            return null;
-        }
-
-        if ($dim === 'array') {
-            return $size;
-        }
-
-        // Get single dimension
-        if (is_numeric($dim)) {
-            return $size[$dim];
-        }
-
-        return "$size[0]x$size[1]";
-    }
-
-    /**
      * Get the width of this image.
      *
      * @return int
      */
     public function getWidth()
     {
-        return $this->getDimensions(0);
+        return $this->getImageBackend()->getWidth();
```

Comment:
This breaks if calling getWidth() on something that's not an image. It should be some empty value.